src/main/java/io/reactivex/processors/MulticastProcessor.java

```diff
@@ -0,0 +1,637 @@
+     * ignores request amounts, uses a fixed buffer
+     * and allows using the onXXX and offer methods
+     * afterwards.
+     */
+    public void start() {
+        if (SubscriptionHelper.setOnce(upstream, EmptySubscription.INSTANCE)) {
+            queue = new SpscArrayQueue<T>(bufferSize);
+        }
+    }
+
+    /**
+     * Initializes this Processor by setting an upstream Subscription that
+     * ignores request amounts, uses an unbounded buffer
+     * and allows using the onXXX and offer methods
+     * afterwards.
+     */
+    public void startUnbounded() {
+        if (SubscriptionHelper.setOnce(upstream, EmptySubscription.INSTANCE)) {
+            queue = new SpscLinkedArrayQueue<T>(bufferSize);
+        }
+    }
+
+    @Override
+    public void onSubscribe(Subscription s) {
+        if (SubscriptionHelper.setOnce(upstream, s)) {
+            if (s instanceof QueueSubscription) {
+                @SuppressWarnings("unchecked")
+                QueueSubscription<T> qs = (QueueSubscription<T>)s;
+
+                int m = qs.requestFusion(QueueSubscription.ANY);
+                if (m == QueueSubscription.SYNC) {
+                    fusionMode = m;
+                    queue = qs;
+                    done = true;
+                    drain();
+                    return;
+                }
+                if (m == QueueSubscription.ASYNC) {
+                    fusionMode = m;
+                    queue = qs;
+
+                    s.request(bufferSize);
+                    return;
+                }
+            }
+
+            queue = new SpscArrayQueue<T>(bufferSize);
+
+            s.request(bufferSize);
+        }
+    }
+
+    @Override
+    public void onNext(T t) {
+        if (once.get()) {
+            return;
+        }
+        if (fusionMode == QueueSubscription.NONE) {
+            if (t == null) {
+                throw new NullPointerException("t is null");
+            }
+            if (!queue.offer(t)) {
+                SubscriptionHelper.cancel(upstream);
+                onError(new MissingBackpressureException());
+                return;
+            }
+        }
+        drain();
+    }
+
+    /**
+     * Tries to offer an item into the internal queue and returns false
+     * if the queue is full.
+     * @param t the item to offer, not null
+     * @return true if successful, false if the queue is full
+     */
+    public boolean offer(T t) {
+        if (once.get()) {
+            return false;
+        }
+        if (t == null) {
+            throw new NullPointerException("t is null");
+        }
+        if (fusionMode == QueueSubscription.NONE) {
+            if (queue.offer(t)) {
+                drain();
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void onError(Throwable t) {
+        if (t == null) {
+            throw new NullPointerException("t is null");
```

Comment:
Didn't we also have another second sentence that was stating that nulls are not allowed per definition?